utils/build-script-impl

```diff
@@ -1035,6 +952,129 @@ function should_build_stdlib_target() {
     fi
 }
 
+function triple_for_stdlib_target() {
+    local stdlib_target=$1
+    case ${stdlib_target} in
+        # FreeBSD
+        freebsd-x86_64)
+            #FIXME: Set a meaningful FreeBSD deployment version
+            FREEBSD_DEPLOYMENT_VERSION="10.3"
```

Comment:
This replaces the FIXME in CMake, about FreeBSD using the local `CMAKE_SYSTEM_VERSION` to derive a deployment target. It should have its own parameter, like Darwin targets do, and that would solve the cross-compiling issue.